src/core_atmosphere/dynamics/mpas_atm_time_integration.F

```diff
@@ -720,11 +723,24 @@ subroutine atm_srk3(domain, dt, itimestep, exchange_halo_group)
       call mpas_timer_start('physics_get_tend')
       rk_step = 1
       dynamics_substep = 1
+
+! apply random perturbation pattern to the tendency
+      tend_names(1) = "rucuten"
+      tend_names(2) = "rvcuten"
+      tend_names(3) = "rublten"
+      tend_names(4) = "rvblten"
+      call stochastic_physics_pattern_apply(domain, 4, tend_names, ierr)
+
       call physics_get_tend( block, mesh, state, diag, tend, tend_physics, &
                              block % configs, rk_step, dynamics_substep, &
                              tend_ru_physics, tend_rtheta_physics, tend_rho_physics, &
                              exchange_halo_group )
       call mpas_timer_stop('physics_get_tend')
+
+! apply random perturbation pattern to the tendency
+      tend_names(1) = "tend_rtheta_physics"
+!      tend_names(2) = "tend_rho_physics"
+      call stochastic_physics_pattern_apply(domain, 1, tend_names, ierr)
```

Comment:
Tendency perturbations are only applied to "rtheta" because it was found that there are no "rho" tendencies, is that correct?

Also, "rtheta" is actually virtual potential temperature, correct?

Reply:
From Ning: "rtheta" is modified virtual potential tempeature which has qv computed and used in its calculation.  "rho" is used in computation of "rtheta" and then discarded, so no need to perturb for SPPT.